R/lint.R

```diff
@@ -335,10 +340,10 @@ validate_linter_object <- function(linter, name) {
     return(linter)
   }
   if (!is.function(linter)) {
-    stop(gettextf(
-      "Expected '%s' to be a function of class 'linter', not a %s of class '%s'",
-      name, typeof(linter), class(linter)[[1L]]
-    ), call. = FALSE)
+    cli_abort(c(
+      i = "Expected {.fn {name}} to be a function of class {.cls linter}.",
+      x = "Instead, it is of class {.cls {class(linter)[[1L]]}}."
```

Comment:
I would assign this outside the error, too many delimiters going on in there:

```r
linter_class <- class(linter)[1L]
... {.cls linter_class}
```

Reply:
actually, elsewhere we just use `class()` inside `{.cls` and {cli} handles it well to give a full class summary:

```r
cli_warn("{.cls {class(Sys.time())}}")
# Warning message:
# <POSIXct/POSIXt>
```

So let's just use that.

Reply:
There is also `{.obj_type_friendly {Sys.time()}}`, but in dev cli, will only show the first class.